src/UI/processmonitor.h

```diff
@@ -47,13 +22,17 @@ class ProcessMonitor : public IProcessMonitor{
     ProcessMonitor(IProcessesSeeker* processesSeeker)
         :processesSeeker{processesSeeker}{};
     void updateProcessesTable() override;
-    ProcessInfo getCopyOfProcessInfoByIndex(const int index) override;
-    ProcessInfo getCopyOfProcessInfoByPid(const int Pid) override;
-    static std::vector<ProcessInfo> mergeProcessesLists(const std::vector<ProcessInfo>& oldProcesses ,
+    ProcessInfo getCopyOfProcessInfoByIndex(const size_t index) const override;
+    ProcessInfo getCopyOfProcessInfoByPid(const DWORD Pid) const override;
+    void setProcessEditableFieldByIndex(const size_t Index, const ProcessEditableFields field, const bool value) override;
+    void setProcessEditableFieldByPid(const DWORD Pid, const ProcessEditableFields field, const bool value) override;
+    size_t getProcessesCount() const override;
+
+    static std::vector<ProcessInfo> mergeProcessesSortedLists(const std::vector<ProcessInfo>& oldProcesses ,
                                                     const std::vector<ProcessInfo>& currentProcesses);
-    void setProcessEditableField(const ProcessEditableFields field) override;
     ~ProcessMonitor();
 private:
+    size_t getIndexOfProcessBySpecyficPid(const DWORD Pid) const;
     std::vector<ProcessInfo> processesInfo;
```

Comment:
Maybe it will be easier to make a map<pid, info> instead of vector to simplify the search.

Reply:
I found out that I need to change 3 classes due to this change , and their unit tests. So I created new class Processes Storage , he takes responsibility about raw storage interaction from Process Monitor.